### PR TITLE
Create platform directory and simplify application startup code

### PR DESCRIPTION
**Pull request by jbalboni**
Connects to https://github.com/department-of-veterans-affairs/vets-contrib/issues/22

Notes:
- I created the platform directory in src because that will cause fewer path changes down the road. We can move it up a directory later (or leave it, depending on how people feel)
- I tried to move as little as possible, but it's still a large-ish PR
- Most of the entry files fit in the simpler startApp mode, some of them do different things with history or routing and drop down a level of abstraction and use the functions that startApp uses.

### CODE REVIEW

File: src/js/auth/auth-entry.jsx [hunk 1]
```diff
@@ -1,27 +1,21 @@
...
-function init() {
-  ReactDOM.render((
-    <Provider store={store}>
-      <Router history={browserHistory} routes={routes}/>
-    </Provider>
-  ), document.getElementById('react-root'));
-}
-
-initReact(init);
+startReactApp(
```
Comment (rianfowler) [inline src/js/auth/auth-entry.jsx]: This is really nice.

File: src/js/vre/chapter36/chapter36-entry.jsx [hunk 2]
```diff
@@ -1,33 +1,14 @@
...
-const store = initCommon(reducer);
-
-const browserHistory = useRouterHistory(createHistory)({
-  basename: manifest.rootUrl
+startApp({
+  url: manifest.rootUrl,
+  reducer,
+  routes
 });
-
```
Comment (ncksllvn) [inline src/js/vre/chapter36/chapter36-entry.jsx]: 🥂

File: src/platform/polyfills/canvas-toBlob.js [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+        const arr = new Uint8Array(len);
+
+        for (let i = 0; i < len; i++) {
+          arr[i] = binStr.charCodeAt(i);
+        }
+
+        callback(new Blob([arr], { type: type || 'image/png' }));
+      });
+    }
+  });
```
Comment (ncksllvn) [inline src/platform/polyfills/canvas-toBlob.js]: Is this used by the picture cropper on VIC I think?

File: src/platform/startup/index.js [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+      </Router>
+    );
+  }
+
+  startReactApp(
+    <Provider store={store}>
+      {content}
+    </Provider>
+  );
+}
```
Comment (ncksllvn) [inline src/platform/startup/index.js]: ✨

**Comment by ncksllvn** [comment]
Looks awesome! Haven't tried running locally yet - is it ready for merge?

File: src/platform/polyfills/canvas-toBlob.js (hunk 3 above)
Comment (rianfowler) [reply src/platform/polyfills/canvas-toBlob.js]: Yes- the `Cropper` component we use does the photo manipulation and display with a canvas.

**Review by rianfowler — APPROVED** [review]
I was able to start, save, and continue an education form with a logged in user locally.

**Comment by jbalboni** [comment]
@ncksllvn I'd like to get a few people to test it, but there are no outstanding changes left for this PR.

File: src/js/letters/routes.jsx [hunk 5]
```diff
@@ -5,29 +5,32 @@ import DownloadLetters from './containers/DownloadLetters.jsx';
 import AddressSection from './containers/AddressSection.jsx';
 import LetterList from './containers/LetterList.jsx';
 import Main from './containers/Main.jsx';
+import LettersApp from './containers/LettersApp.jsx';
 
-const routes = [
-  <Route
-    component={Main}
-    key="main">
+const routes = (
```
Comment (jbalboni) [inline src/js/letters/routes.jsx]: FYI @cvalarida

**Pull request merged**